### ISSUE TITLE
Testing resources with optional parameters

### ISSUE THREAD

**Issue by imp-partner-test** [apiaryio/dredd#88]
How do I test optional parameters?
----Sample URL----
Resources/CarSharing/All?lat={lat}&lng={lng}&radius={radius}
----Parameters description----
- Parameters
  - lat (optional, string, `52.52`) ... String `lat` latitude value. Has example value.
  - lng (optional, string, `13.39`) ... String `lng` longitude value. Has example value.
  - radius (optional, string, `1000`) ... String `radius` longitude value. Has example value.

I would like to check at least 2 cases: 1) a call with parameters and their sample values and 2) a call without any parameters:
1) Resources/CarSharing/All?lat=52.52&lng=13.39&radius=1000
2) Resources/CarSharing/All

Thank you!

**Comment by netmilk** [comment]
Just a little update:
You can specify two [different examples (implicit requests and response pairs](https://github.com/apiaryio/api-blueprint/blob/master/API%20Blueprint%20Specification.md#example-multiple-transaction-examples) for the same action, and then modify request parameters in the before hook very similar way as in [this example](https://github.com/apiaryio/dredd/issues/108#issue-45550738).

More straightforward syntax for this is discussed also in API Blueprint project: https://github.com/apiaryio/api-blueprint/issues/132 https://github.com/apiaryio/api-blueprint/issues/58#issuecomment-34958639

**Comment by netmilk** [comment]
This is blocked by support for Parameters under each Request secction in the API blueprint https://github.com/apiaryio/api-blueprint/issues/58

At this moment, as a workaround, you can change the request URI in the `before` hook and describe used parameters in the Request description section.

**Comment by hendrikmaus** [comment]
:+1:

**Comment by mhueter** [comment]
+1

**Comment by pakore** [comment]
+1. 4 years since this and it's a pretty basic functionality IMHO...

**Comment by honzajavorek** [comment]
@pakore 

> This is blocked by support for Parameters under each Request secction in the API Blueprint https://github.com/apiaryio/api-blueprint/issues/58

The implementation in the API Blueprint format still isn't in general availability. Please express your feelings under the linked issue. Dredd isn't in a position it could easily overcome the problem, as it is a limitation of the format.

**Comment by honzajavorek** [comment]
@pakore I'm sorry, I did not notice the issue is locked. My apologies. It is [the first thing on the public roadmap](https://github.com/apiaryio/api-blueprint/wiki/Roadmap) though.

**Comment by honzajavorek** [comment]
Even though it's not an officially rolled-out feature in API Blueprint and Apiary, the parser supports request parameters for a long time already and Dredd does as well. Please open other issues if you find the behavior buggy. Closing this in favor of https://github.com/apiaryio/dredd/issues/1118.
